src/Session.cpp

```diff
@@ -92,14 +93,32 @@ Session::Session(const Session &s)
     clean();
 }
 
+void Session::deepCopyUsers(const std::unordered_map<std::string, User*>& newUsers)
```

Comment:
change function name to deepCopyMap
there's nothing special about users

Reply:
But this function specifically works for the userMap. It doesn't implement some high level deep copy for generic maps. It accesses `this->userMap`

Reply:
I agree, but why not doing it generic as in copy vector?

Reply:
Because then I need to pass in more parameters and define the function as a two class template (for the first and second of each pair)